service/controller/key/key.go

```diff
@@ -387,7 +387,7 @@ func MasterNatGatewayID(cr providerv1alpha1.AzureConfig, subscriptionID string)
 
 // MasterNICName returns name of the master NIC.
 func MasterNICName(customObject providerv1alpha1.AzureConfig) string {
-	return fmt.Sprintf("%s-Master-1-NIC", ClusterID(&customObject))
+	return fmt.Sprintf("%s-master-%s-nic", ClusterID(&customObject), ClusterID(&customObject))
```

Comment:
if we change the name, is it backwards compatible? can customers upgrade?

Reply:
we didn't change the name. The function was unused. I needed it so I fixed it (the name returned by the function didn't match the actual name in the ARM template) and used it